website/docs/how-to/define-api-in-typescript.md

```diff
@@ -2,3 +2,197 @@
 title: How To Define a Portal APIs
 sidebar_label: Define your own Portal APIs
 ---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+One of the biggest benefits of including Ionic Portals in an application is the ability to easily communicate between web and native code using the [PortalsPlugin](../reference/web/portals-plugin). However, in some more niche cases, creating your own Plugins may be neccessary. By creating a [Capacitor Plugin](https://capacitorjs.com/docs/plugins/creating-plugins), you can create your own API to communicate between web and native code.
+
+For this example, we will create a Plugin called `EchoPlugin` that has a single function: `echo`.
+
+## Creating API definitions
+
+We strongly recommend using TypeScript to create a type defintion file which can be used to define your API. This way, there is a central source of truth for the API across Android and iOS as well as having type defintions for the web code.
+
+```typescript
+export interface IEchoPlugin {
+   echo(options: { value: string }): Promise<{ value: string }>;
+}
+```
+
+On the Android or iOS side, the `EchoPlugin` class will need to match this type signature.
```

Comment:
Technically not correct. The swift/java classes have method names that match up, but their parameters and return types aren't the same as the interface. Maybe something like:

"On the Android and iOS side, the `EchoPlugin` class will implement methods that listen for calls made to methods on the `EchoPlugin` interface."